src/main/python/ttconv/srt/writer.py

```diff
@@ -174,26 +174,24 @@ def __str__(self) -> str:
 #
 
 
-def from_model(doc: model.ContentDocument) -> str:
+def from_model(doc: model.ContentDocument, progress_callback=lambda _: None) -> str:
   """Converts the data model to a SRT document"""
 
   srt = SrtContext()
 
   # Compute ISDs
 
-  isds = ISD.generate_isd_sequence(doc)
+  isds = ISD.generate_isd_sequence(doc, progress_callback)
 
   # process ISDs
 
-  for offset, isd in isds:
+  for i, (offset, isd) in enumerate(isds):
 
     for srt_filter in srt.filters:
       srt_filter.process(isd)
 
     srt.add_isd(isd, offset)
 
-    progress_callback((index + 1) / nb_significant_times)
```

Comment:
It means that the "writing" progress is only defined by the ISD generation? Shouldn't we share the percentage of progress between ISD and SRT generations (70/30 for instance)?

Reply:
Done at https://github.com/sandflow/ttconv/pull/160/commits/ced81f604b69004903af94b587da33ce60094c35